moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java

```diff
@@ -682,6 +670,65 @@ public void setupVersionHeader() {
         restClient.setHeader("Version", authenticator.getApiMajorVersion());
     }
 
+    private void setPersistentV3AuthHeaders() {
+        restClient.setHeader(SESSION_TTL, "120"); // 2h
+        restClient.setHeader(PREFER, "persistent-auth, csrf-protection");
+    }
+
+    private void resetClientSettings() {
+        restClient.setHeader(SESSION_TTL, "");
+        restClient.setHeader(PREFER, "");
+        restClient.setCookie(JSESSIONID, "");
+        restClient.setAuthentication(new HttpAuthentication() {
+            @Override
+            public String getHeaderValue() {
+                // empty authentication - e.g. not the basic one
+                return "";
+            }
+        });
+    }
+
+    private void updateClientBeforeCall() {
+        restClient.setHeader(FILTER, Boolean.toString(!authenticator.hasAdminPermissions()));
+        requestFactory.setCertificateHandlingMode(authenticator.getCertHandlingStrategy());
+        restClient.setRootUrl(authenticator.getApiUrl());
+    }
+
+    /**
+     * @param username username
+     * @param password password
+     * @return auth token depending on API version, if enforceBasicAuth is true then returns empty string
+     */
+    public String login(String username, String password) {
+        resetClientSettings();
+        updateClientBeforeCall();
+        restClient.setHttpBasicAuth(username, password);
+
+        Api api = restClient.loginV3();
```

Comment:
@suomiy you can remove it completely - I have introduced it only as a fallback because I did not really trust my code which was handling the sessionId :) But now I think it is not needed anymore, so it can be removed.